frontend/src/component/common/StaleDataNotification/StaleDataNotification.tsx

```diff
@@ -0,0 +1,75 @@
+import { Typography, Button, useTheme, useMediaQuery } from '@mui/material';
+import EventDiff from 'component/events/EventDiff/EventDiff';
+import { useThemeStyles } from 'themes/themeStyles';
+import AnimateOnMount from '../AnimateOnMount/AnimateOnMount';
+
+interface IStaleDataNotification {
+    refresh: () => void;
+    afterSubmitAction: Function;
+    data: unknown;
+    cache: unknown;
+    show: boolean;
+}
+
+export const StaleDataNotification = ({
+    refresh,
+    show,
+    afterSubmitAction,
+    data,
+    cache,
+}: IStaleDataNotification) => {
+    const { classes: themeStyles } = useThemeStyles();
+    const theme = useTheme();
+    const smallScreen = useMediaQuery(`(max-width:${600}px)`);
+
+    const getStyles = () => {
+        const base = {
+            padding: `${theme.spacing(3)} ${theme.spacing(4)}`,
+            boxShadow: theme.boxShadows.elevated,
+            borderRadius: theme.shape.borderRadiusLarge,
+            backgroundColor: theme.palette.background.paper,
+            maxWidth: '600px',
+            zIndex: 1000,
+        };
+        if (smallScreen) {
+            return {
+                ...base,
+                right: 0,
+                left: 0,
+                bottom: 0,
+                borderRadius: 0,
+            };
+        }
+        return base;
+    };
+
+    return (
+        <AnimateOnMount
+            mounted={show}
+            start={themeStyles.fadeInBottomStartWithoutFixed}
+            enter={themeStyles.fadeInBottomEnter}
+            style={getStyles()}
+        >
+            <Typography variant="h5" sx={{ my: 2, mb: 2 }}>
```

Comment:
We might want to follow a StyledComponent approach by default instead of using `sx`, even though for things like these it's more convenient.

Reply:
I think it's a bit overkill to use a StyledComponent for these kind of styling changes. It's hard to create an abstraction that makes sense to generalise, because it depends on the situation it's used.